File: src/main/java/com/ekart/model/Card.java
package com.ekart.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
public class Card {

  private String cardNumber;
  private String name;
  private LocalDate validTo;

  private double balance;

  public boolean charge(double total)
  {
    if (total < 0)
      throw new IllegalArgumentException("Total charge Must be non-negative");

    if (total > balance)
      return false;

    balance -= total;
    return true;
  }

  public void addCredit(double credit)
  {
    if (credit < 0)
      throw new IllegalArgumentException("Credit must be non-negative");

      balance += credit;
  }

}

File: src/main/.net/Model/Card.cs
namespace ShoppingApp
{
    public sealed class Card
    {
        public string CardNumber { get; }
        public string Name { get; }
        public DateTime ValidTo { get; }

        private double _balance;
        public double Balance => _balance;

        public Card(string cardNumber, string name, DateTime validTo, double balance)
        {
            CardNumber = cardNumber;
            Name = name;
            ValidTo = validTo;
            _balance = balance;
        }

        public bool Charge(double total)
        {
            if (total < 0)
                throw new ArgumentException("Must be non-negative", nameof(total));

            if (total > _balance)
                return false;

            _balance -= total;
            return true;
        }

        public void AddCredit(double credit)
        {
            if (credit < 0)
                throw new ArgumentException("Must be non-negative", nameof(credit));

            _balance += credit;
        }
    }
}